### DISCUSSION TITLE
Heyy can you guys suggest free and efficient APIs for projects such as image generation, image captioning etc?

### DISCUSSION THREAD

**Discussion by chriz-ty** [community/community#144183] [Apps, API and Webhooks] [answered]
### APIs

Question

### Body

I'm planning to do a project to include the features like image generation and image captioning etc. So, i need some free API sources.. also, does anyone familiar with NLP based free APIs for text translation?????

**Comment by samuel-404** [comment, marked as answer]
1. OpenAI DALL-E for Image Generation
2.  Hugging Face Transformers and Datasets
3. DeepAI Image APIs
4.  Google Vision AI
5. Imagga
